fluss-flink/fluss-flink-common/src/test/java/com/alibaba/fluss/flink/source/deserializer/FlussRowDeserializerTest.java

```diff
@@ -0,0 +1,154 @@
+        row.setField(2, 45);
+        row.setField(3, BinaryString.fromString("Test addr"));
+
+        ScanRecord scanRecord = new ScanRecord(row);
+
+        RowDataDeserializationSchema deserializer = getRowDataDeserializationSchema(rowType);
+        RowData result = deserializer.deserialize(scanRecord);
+
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(row, scanRecord.getRow());
+    }
+
+    private @NotNull RowDataDeserializationSchema getRowDataDeserializationSchema(RowType rowType)
+            throws Exception {
+        RowDataDeserializationSchema deserializationSchema = new RowDataDeserializationSchema();
+        deserializationSchema.open(
+                new FlussDeserializationSchema.InitializationContext() {
+                    @Override
+                    public MetricGroup getMetricGroup() {
+                        return null;
+                    }
+
+                    @Override
+                    public UserCodeClassLoader getUserCodeClassLoader() {
+                        return null;
+                    }
+
+                    @Override
+                    public RowType getRowSchema() {
+                        return rowType;
+                    }
+                });
+        return deserializationSchema;
+    }
+
+    @Test
+    public void testDeserializeWithNullRecord() {
+        assertThrows(NullPointerException.class, () -> schema.deserialize(null));
+    }
+
+    @Test
+    public void testGetProducedType() {
+        TypeInformation<RowData> typeInfo = schema.getProducedType();
+
+        assertNotNull(typeInfo);
+        assertEquals(RowData.class, typeInfo.getTypeClass());
+    }
+
+    @Test
+    public void testSerializable() throws Exception {
+        // Serialize
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ObjectOutputStream oos = new ObjectOutputStream(baos);
+        oos.writeObject(schema);
+        oos.close();
+
+        // Deserialize
+        ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
+        ObjectInputStream ois = new ObjectInputStream(bais);
+        RowDataDeserializationSchema deserializedSchema =
+                (RowDataDeserializationSchema) ois.readObject();
+        ois.close();
+
+        // Verify
+        assertNotNull(deserializedSchema);
+        assertNotNull(deserializedSchema.getProducedType());
+        assertEquals(schema.getProducedType(), deserializedSchema.getProducedType());
+    }
+
+    @Test
+    public void testDifferentRowTypes() throws Exception {
+        // Test with different row types
+        List<DataField> simpleFields = Arrays.asList(new DataField("id", DataTypes.BIGINT()));
+
+        RowDataDeserializationSchema simpleSchema =
+                getRowDataDeserializationSchema(new RowType(simpleFields));
+
+        assertNotNull(simpleSchema);
+        assertEquals(RowData.class, simpleSchema.getProducedType().getTypeClass());
```

Comment:
assert the `typeInfo.createSerializer(new ExecutionConfig())` is an instance of `RowDataSerializer`